aep/general/0137/aep.md.j2

```diff
@@ -0,0 +1,143 @@
+# Apply
+
+In REST APIs, it is customary to make a `PUT` request to a collection's URI (for
+example, `/v1/publishers/{publisher}/books`) in order to create or update a new
+resource within that collection.
+
+Resource-oriented design (AEP-121) honors this pattern through the `Apply`
+method. These RPCs accept the parent collection and the resource to create or
+update, and return the final resource.
+
+## Guidance
+
+APIs **should** provide an apply method for resources unless it is not valuable
+for users to do so.
+
+### Operation
+
+Apply methods are specified using the following pattern:
+
+- The HTTP verb **must** be `PUT`.
+- Some resources take longer to be applied than is reasonable for a regular API
+  request. In this situation, the API **should** use a [long-running
+  operation](/long-running-operations).
```

Comment:
Maybe we should also answer this for long-running updates...but what happens to the resource is the interim?  Is it write-locked?

Hmm...if I have long-running updates but sync/immediate applies, what happens if I try to apply during a long-running update?

Reply:
It's a good question... 

I don't think we can make "write lock for concurrent updates" a *must* - that's difficult behavior to implement server-side, and we may end up removing a lot of APIs from being AEP-compliant if we have a hard stance on that. 

I do think we have the ability to get some of these patterns though. E.g. for those APIs that do need some level of check-and-set, we do have https://aep.dev/154/. 

I think if we did tackle concurrency, we might be better off filling out https://aep.dev/151/ some more.

WDYT about me starting an issue and we can follow up? aep.dev/134 doesn't have speak to this either.

Reply:
Sure, starting an issue to unblock the PR SGTM.

Reply:
https://github.com/aep-dev/aeps/issues/250